openest/generate/functions.py

```diff
@@ -443,6 +443,148 @@ def describe():
                     description="Sum the results of multiple previous calculations.")
 
 
+
+class FractionSum(calculation.Calculation):
+    """Sum of two subcalculations weighted on fraction of unity.
+
+    In short, ``frac * subcalc[0] + (1 - frac) * subcalc[1]``.
+
+    Parameters
+    ----------
+    frac : ``openest.generate.calculation.Calculation``
+        Result gives weight of first subcalculation. 1 - `frac` is weight of
+        second subcalculation. Must be ≤ 1. The
+    subcalcs : Sequence of ``openest.generate.calculation.Calculation``
+        Calculations to weight. Must be ``len(subcalcs) == 2``. The units of
+        the subcalcs elements must be the same.
+    unshift : bool, optional
+    """
+
+    def __init__(self, frac, subcalcs, unshift=True):
+        if len(subcalcs) != 2:
+            raise ValueError('Length of `subcalcs` must be 2')
+
+        fullunitses = subcalcs[0].unitses[:]
+        for ii in range(1, len(subcalcs)):
+            assert subcalcs[0].unitses[0] == subcalcs[ii].unitses[0], "%s <> %s" % (subcalcs[0].unitses[0], subcalcs[ii].unitses[0])
+            fullunitses.extend(subcalcs[ii].unitses)
```

Comment:
Also check that frac is unitless.

Reply:
@jrising What are accepted values for unitless units?

Reply:
Resolved with ec0e39203139fd0632a354e398e6075aea9f8cfe